frontend/public/components/overview/index.tsx

```diff
@@ -243,27 +279,41 @@ const OverviewHeading: React.SFC<OverviewHeadingProps> = ({disabled, groupOption
               style={{display: 'inline-block'}}
               title={selectedGroup}
             />
+          </div> }
+          <div className="form-group overview-toolbar__form-group">
+            <TextFilter
+              autoFocus={!disabled}
+              defaultValue={''}
+              disabled={disabled}
+              label="Resources by name"
+              onChange={handleFilterChange}
+            />
           </div>
-        }
-        <div className="form-group overview-toolbar__form-group">
-          <TextFilter
-            autoFocus={!disabled}
-            defaultValue={''}
-            disabled={disabled}
-            label="Resources by name"
-            onChange={handleFilterChange}
-          />
-        </div>
+        </React.Fragment>}
+        {selectedView === View.Dashboard && !_.isEmpty(project) && <div className="form-group overview-toolbar__form-group">
+          <ActionsMenu actions={overviewMenuActions.map((a: CogAction) => a(ProjectModel, project))} />
+        </div>}
       </Toolbar.RightContent>
     </Toolbar>
   </div>
 );
 
-class OverviewDetails extends SafetyFirst<OverviewDetailsProps, OverviewDetailsState> {
+const OverviewHeading = connect<OverviewHeadingPropsFromState, OverviewHeadingPropsFromDispatch, OverviewHeadingOwnProps>(headingStateToProps, headingDispatchToProps)(OverviewHeading_);
+
+const mainContentStateToProps = ({UI}): OverviewMainContentPropsFromState => {
+  const selectedView = UI.getIn(['overview', 'selectedView'], View.Resources);
+  return { selectedView };
+};
+
+const mainContentDispatchToProps = (dispatch): OverviewMainContentPropsFromDispatch => ({
+  updateResources: (items: any[]) => dispatch(UIActions.updateOverviewResources(items)),
+});
+
+class OverviewMainContent_ extends SafetyFirst<OverviewMainContentProps, OverviewMainContentState> {
   /* eslint-disable-next-line no-undef */
   metricsInterval_: any;
 
-  constructor(props: OverviewDetailsProps) {
+  constructor(props: OverviewMainContentProps) {
```

Comment:
No change necessary for this PR, but I think I got a little over-zealous with the type definitions here. These would be covered by the class definition above.